Line limits assertion fix

nkorinek opened the pull request:
Changed how the limits of a plotted line are checked to make it more generous for small mistakes.
addresses: https://github.com/earthlab/matplotcheck/issues/235

nkorinek committed:
Merge pull request #1 from earthlab/master

Updating files

pyup-bot committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

Update setuptools from 42.0.2 to 45.2.0 (#190)

raster inherits from vector rather than base (#76)

Allow geodataframes in assert_polygons (#188)

fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

update changelog for release

Bump version: 0.1.1 → 0.1.2

Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Update codecov to 2.0.16 (#202)

* Update codecov from 2.0.15 to 2.0.16

* update from master (#211)

* Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>
Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

Update setuptools from 45.2.0 to 46.0.0 (#215)

Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

Update pytest from 5.3.5 to 5.4.1

merge

Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Add tests to the Autograde module (#224)

* Added tests for the autograde module

* removed unneccesary import

* Fixing issues that were failing in base_data

* codacy

* flake8

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Updated changelog

* flake8 fix

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Hist bin midpoints (#204)

* Add a get and assert function for bin midpoints, as well as tests for those functions

* Updated docstrings

* Added changelog changes and fixed minor formatting issue

* small flake8 fix

* Added midpoints functionatlity to the vignette for histogram testing

* Adding in changes suggested on GitHub1

* Fixed function description

* Took out all instances of  in example

* update docs

* Fixed small bug with section titles

* Update matplotcheck/tests/test_base_data.py

* reworded changelog

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Update setuptools from 46.1.1 to 46.1.3 (#231)

Title assert accepts strings (#229)

* Fixed assert_title_contains to take strings as well as lists

* Better implementation

* Changelog update

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

M2r remove (#247)

* Removed m2r from matplotcheck, and reformated code of conduct

* Seeing if rebase works

* removed an m2r import

* Fixing up the docs a bit

* changelog updates

Allow geodataframes in assert_polygons (#188)

fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Get images function (#193)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Bump version: 0.1.1 → 0.1.2

Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

# This is the commit message #2:

Update setuptools from 46.1.1 to 46.1.3 (#231)

pyup-bot committed:
Update setuptools from 42.0.2 to 45.2.0 (#190)

nkorinek committed:
Merge pull request #2 from earthlab/master

Update setuptools from 42.0.2 to 45.2.0 (#190)

Commit:
raster inherits from vector rather than base (#76)

nkorinek committed:
Merge pull request #3 from earthlab/master

raster inherits from vector rather than base (#76)

nkorinek committed:
Allow geodataframes in assert_polygons (#188)

nkorinek committed:
Merge pull request #4 from earthlab/master

Allow geodataframes in assert_polygons (#188)

Commit:
fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

nkorinek committed:
Merge pull request #5 from earthlab/master

Updating from master

Commit:
update changelog for release

Commit:
Bump version: 0.1.1 → 0.1.2

nkorinek committed:
Merge pull request #7 from earthlab/master

updating

nkorinek committed:
Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

nkorinek committed:
Merge pull request #8 from earthlab/master

Add flake8 fix [WIP] (#199)

nkorinek committed:
Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

nkorinek committed:
Merge pull request #9 from earthlab/master

Assert string accept spaces (#205)

pyup-bot committed:
Update codecov to 2.0.16 (#202)

* Update codecov from 2.0.15 to 2.0.16

* update from master (#211)

* Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>
Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

nkorinek committed:
Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

pyup-bot committed:
Update setuptools from 45.2.0 to 46.0.0 (#215)

nkorinek committed:
Merge pull request #10 from earthlab/master

update

pyup-bot committed:
Update pytest from 5.3.5 to 5.4.1

nkorinek committed:
Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

nkorinek committed:
Merge pull request #11 from earthlab/master

Add Assert points function & cleanup duplicate methods (#203)

nkorinek committed:
Merge pull request #12 from earthlab/master

Update pytest from 5.3.5 to 5.4.1

nkorinek committed:
Merge pull request #13 from earthlab/master

update

nkorinek committed:
Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

nkorinek committed:
Merge pull request #14 from earthlab/master

update

nkorinek committed:
Add tests to the Autograde module (#224)

* Added tests for the autograde module

* removed unneccesary import

* Fixing issues that were failing in base_data

* codacy

* flake8

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Updated changelog

* flake8 fix

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

nkorinek committed:
Merge pull request #15 from earthlab/master

Add tests to the Autograde module (#224)

nkorinek committed:
Hist bin midpoints (#204)

* Add a get and assert function for bin midpoints, as well as tests for those functions

* Updated docstrings

* Added changelog changes and fixed minor formatting issue

* small flake8 fix

* Added midpoints functionatlity to the vignette for histogram testing

* Adding in changes suggested on GitHub1

* Fixed function description

* Took out all instances of  in example

* update docs

* Fixed small bug with section titles

* Update matplotcheck/tests/test_base_data.py

* reworded changelog

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

pyup-bot committed:
Update setuptools from 46.1.1 to 46.1.3 (#231)

nkorinek committed:
Merge pull request #16 from earthlab/master

update

nkorinek committed:
Title assert accepts strings (#229)

* Fixed assert_title_contains to take strings as well as lists

* Better implementation

* Changelog update

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

nkorinek committed:
Merge pull request #17 from earthlab/master

Title assert accepts strings (#229)

nkorinek committed:
Fixed bug with how line limits are checke

nkorinek committed:
M2r remove (#247)

* Removed m2r from matplotcheck, and reformated code of conduct

* Seeing if rebase works

* removed an m2r import

* Fixing up the docs a bit

* changelog updates

nkorinek committed:
Merge pull request #18 from earthlab/master

M2r remove (#247)

nkorinek committed:
merge

nkorinek committed:
merge

nkorinek committed:
Updating in order to squash commits!

lwasser commented:
hey @nkorinek looks like we have some conflicts now. also i'm seeing build errors. is pillow installed in your dev envt? it seems like it's missing from the travis errors. i'm curious whether we should add it as a dev requirement?

lwasser commented:
actually maybe it's the same m2r issue??

nkorinek committed:
merge

nkorinek commented:
@lwasser I've updated it to include the m2r fix. Hopefully that fixes the issues you were seeing

lwasser commented:
great. thank you @nkorinek !!

Commit:
fix copyright on docs (#244)

Commit:
Add pillow (#254)

* add pillow as a dev requirement

* update for pillow addition

Commit:
Merge branch 'master' into regression-asssert

Commit:
Merge branch 'master' into regression-asssert

lwasser commented:
@nkorinek are you still working on this guy? is it ready to merge if CI passes or is it a WIP?

nkorinek commented:
@lwasser fixing a bug with this right now, once I push up the changes it should be good to go

nkorinek committed:
Fixed issue with new way of checking that the data is covered by the line

nkorinek committed:
Changelog update

nkorinek committed:
merge

nkorinek commented:
@lwasser did more testing on this and it should be good to merge once that issue with CI is fixed

lwasser commented:
thank you @nkorinek i'm going to give filipe a day or two to respond... and if he doesn't i'll come up with plan b :) but i suspect he will give me some suggestions regarding things to try!!

nkorinek committed:
Merge

nkorinek committed:
Merge branch 'earthlab-master'

nkorinek committed:
Changelog to rst (#267)

* Update setuptools from 42.0.2 to 45.2.0 (#190)

* raster inherits from vector rather than base (#76)

* Allow geodataframes in assert_polygons (#188)

* fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

* update changelog for release

* Bump version: 0.1.1 → 0.1.2

* Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

* Update codecov to 2.0.16 (#202)

* Update codecov from 2.0.15 to 2.0.16

* update from master (#211)

* Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>
Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

* Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

* Update setuptools from 45.2.0 to 46.0.0 (#215)

* Update pytest from 5.3.5 to 5.4.1

* Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

* merge

* Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Add tests to the Autograde module (#224)

* Added tests for the autograde module

* removed unneccesary import

* Fixing issues that were failing in base_data

* codacy

* flake8

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Updated changelog

* flake8 fix

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Hist bin midpoints (#204)

* Add a get and assert function for bin midpoints, as well as tests for those functions

* Updated docstrings

* Added changelog changes and fixed minor formatting issue

* small flake8 fix

* Added midpoints functionatlity to the vignette for histogram testing

* Adding in changes suggested on GitHub1

* Fixed function description

* Took out all instances of  in example

* update docs

* Fixed small bug with section titles

* Update matplotcheck/tests/test_base_data.py

* reworded changelog

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Update setuptools from 46.1.1 to 46.1.3 (#231)

* Title assert accepts strings (#229)

* Fixed assert_title_contains to take strings as well as lists

* Better implementation

* Changelog update

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* M2r remove (#247)

* Removed m2r from matplotcheck, and reformated code of conduct

* Seeing if rebase works

* removed an m2r import

* Fixing up the docs a bit

* changelog updates

* # This is a combination of 2 commits.
# This is the 1st commit message:

Update setuptools from 42.0.2 to 45.2.0 (#190)

raster inherits from vector rather than base (#76)

Allow geodataframes in assert_polygons (#188)

fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

update changelog for release

Bump version: 0.1.1 → 0.1.2

Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Update codecov to 2.0.16 (#202)

* Update codecov from 2.0.15 to 2.0.16

* update from master (#211)

* Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

* Assert string accept spaces (#205)

* Made it so that assert_string_contains accepts key words with spaces

* black

* renaming function

* Showed assert_title_contains working with a space in the keyword in vignette

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>
Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

Update setuptools from 45.2.0 to 46.0.0 (#215)

Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

Update pytest from 5.3.5 to 5.4.1

merge

Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Add tests to the Autograde module (#224)

* Added tests for the autograde module

* removed unneccesary import

* Fixing issues that were failing in base_data

* codacy

* flake8

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Update matplotcheck/tests/test_autograde.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* Updated changelog

* flake8 fix

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Hist bin midpoints (#204)

* Add a get and assert function for bin midpoints, as well as tests for those functions

* Updated docstrings

* Added changelog changes and fixed minor formatting issue

* small flake8 fix

* Added midpoints functionatlity to the vignette for histogram testing

* Adding in changes suggested on GitHub1

* Fixed function description

* Took out all instances of  in example

* update docs

* Fixed small bug with section titles

* Update matplotcheck/tests/test_base_data.py

* reworded changelog

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Update setuptools from 46.1.1 to 46.1.3 (#231)

Title assert accepts strings (#229)

* Fixed assert_title_contains to take strings as well as lists

* Better implementation

* Changelog update

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

M2r remove (#247)

* Removed m2r from matplotcheck, and reformated code of conduct

* Seeing if rebase works

* removed an m2r import

* Fixing up the docs a bit

* changelog updates

Allow geodataframes in assert_polygons (#188)

fixing a few syntax errors (#194)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* fixing a few syntax errors

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* remove tabls

Co-authored-by: Nathan Korinek <nako1890@colorado.edu>

Get images function (#193)

* First draft of get_images function

* Added a test for get_image

* Changelog update

* minor formatting change.

* black changes

* Update matplotcheck/raster.py

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Bump version: 0.1.1 → 0.1.2

Add flake8 fix [WIP] (#199)

* add flake 8

* yay cleanup

* update change log

* add flake 8 dep

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* more line length issues

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* fixed test_base_axis.py

* fixed text_base_data.py

* fixed test_base_legends.py

* fixed test_base_titles_captions.py

* Format files to be flake8 acceptable (#197)

* autograde flake8 fixes

* Fixed base to be flake8 compliant (phew)

* Made folium.py flake8 compliant

* made notebook.py flake8 compliant

* Fixed raster.py for flake8

* Fixed timeseries.py

* fixed conftest.py

* fixed test_base.py

* Fixed test_raster.py and issue with legends tests

* black

* more black changes

* Fixing merge conflicts that were found

* Changing accepted changes

* minor changes

* Changing to make CI happy, modified to pass black and make -B docs

* Reformatting to make black and flake8 happy. Also updated crs assignment

* Fixed formatting issue with timeseries

* Better fix for timeseries formatting

* Black

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Contributors update (#213)

* Added contributor rst file and moved that information out of the README file

* Implemented requested changes to README

Add Assert points function & cleanup duplicate methods (#203)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* Taking out redundant tests

* Typo

* Fixing how vector checks for truth value of a dataframe

Additional vector tests (#212)

* Added a get_points() and assert_points() function to the vector tester.

* Added in get_points() and assert_points() functions with tests

* Added proper documentation

* Small fix to please codacy

* black

* Updated changelog

* First round of tests for vector

* black

* Rough draft for bug fix

* Fixed bug with multiple geometries plotted alongside bug with identical x values causing failure!

* typo

* Fixed small bug with markersize

* Added comments explaining code

* rough drafts of more tests

* Added more tests for legends

* Added more tests, and took out broken tests

* small codacy fix

* Fixed test!

* Taking out redundant tests

* Took out unneccesary tests

* Added plt.close

* Added more tests

* Typo

* more tests for uncovered parts of the vector file

* black

* Fixed issues with vector checking truth value of dataframe, and added tests

* Fixing how vector checks for truth value of a dataframe

* Added more coverage!

* Broke tests up into individual files

* black

* Added tests for outlier cases

* Update matplotcheck/tests/test_points.py

typo

Co-Authored-By: Leah Wasser <leah.wasser@colorado.edu>

* took plt.gca() out of tests

* Added changes suggested on GitHub

* Update CHANGELOG.md

* fix import order

* import order fix

* import order

* import order

Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

# This is the commit message #2:

Update setuptools from 46.1.1 to 46.1.3 (#231)

* fix copyright on docs (#244)

* Add pillow (#254)

* add pillow as a dev requirement

* update for pillow addition

* Changed the changelog to an rst file instead of a markdown file and added it to the docs build.

* Put changelog changes into the new changelog lol

Co-authored-by: pyup.io bot <github-bot@pyup.io>
Co-authored-by: Leah Wasser <leah.wasser@colorado.edu>

Commit:
setup greetings! (#257)

lwasser commented on matplotcheck/base.py:
```diff
@@ -1102,7 +1102,13 @@ def assert_line(
                 flag_exist = True
                 line_x_vals = [coord[0] for coord in path_verts]
-                if min(line_x_vals) <= min_val and max(line_x_vals) >= max_val:
+                if (
```
@nkorinek can you please add a comment here that explains how this fixes this issue. i understand the issue at a high level. so what if the data have a few extreme edge cases outlier points. and the regression line only fits to 3/4 of the data or 1/2 of the data. will this fail? i think i just need a bit more clarification 

did we discuss making the data limits of a line optional? that could be something else to consider OR creating a regression line and provided it to the method. i just want to talk this through a bit more so i understand and i want the code to have a comment for the same reason. thank you again!

lwasser reviewed (changes requested):
i have a question about how the data lims are tested here! ALso we have change log issues now. as always :)

lwasser commented:
ok this guy still needs a bit of love @nkorinek before i can merge it. see my questions above.  i know i need to merge this prior to merging the lines vignette so let's focus on this guy first. 
update from master and then ofcourse the change log needs an update (md vs rst)

nkorinek commented:
@lwasser oh sorry! I added a comment in the code, but I'll clarify here as well and see if this is what we want to do. 

> so what if the data have a few extreme edge cases outlier points. and the regression line only fits to 3/4 of the data or 1/2 of the data. will this fail? 

This will not fail in this case. In fact most of the time the line does not cover the entirety of the limits, and will be within the bounds but not close to the x and y limits. It seems to do this on lines that should pass normally, however, so maybe that's why this was done this way before? Originally I had this JUST check if the limits were close to the expected limits, but that broke a lot of the tests, so if it checks that it's either within the limits or very close to the limits that's how I got around that issue. Checking that the line was within the limits is how the check was run before. We could change this check if we wanted to!  

> did we discuss making the data limits of a line optional? that could be something else to consider

I don't remember discussing this, but we may have discussed it and I just forgot. Either way, it would be fairly simple to add, but I'm not sure why you would turn that check off. The only time I would turn that check off is if the check was failing and it wasn't meant to fail, but hopefully this PR fixes that. Let me know what you think!

nkorinek committed:
update

nkorinek committed:
Added comment to clarify changes

lwasser commented:
let's just add a few tests to ensure this works in all combinations of points + outlier points

nkorinek commented:
@lwasser this can be closed due to this pr: #274
